### PR TITLE
Drop cuda 10.x support.

### PR DESCRIPTION
**Pull request by qiuxiafei**
As TensorFlow master has promoted C++ standard to C++17, which is out of support capability of nvcc 10.x. Meanwhile, DISC compiler is still built within TensorFlow workspace. There will be heavy engineering burden to maintain backward compatibility of TensorFlow codebase.
After offline discussion, we decide to drop support of CUDA 10.x officially in open-source repository.

### CODE REVIEW

**Comment by Yancey1989** [comment]
also need to update the README: https://github.com/alibaba/BladeDISC#building-status

File: .github/workflows/pytorch181_gpu.yml [hunk 1]
```diff
@@ -1,25 +0,0 @@
-name: pytorch1_8_1-cuda10_2
-on:
```
Comment (fortianyou) [inline .github/workflows/pytorch181_gpu.yml:2]: Can we add a CI for 'pytorch1_8_1-cu113` or other version of cuda 11?

Comment (qiuxiafei) [reply .github/workflows/pytorch181_gpu.yml:2]: Of cause but not in this PR?

Comment (qiuxiafei) [reply .github/workflows/pytorch181_gpu.yml:2]: https://github.com/alibaba/BladeDISC/issues/513

**Review by fortianyou — APPROVED** [review]
LGTM

**Pull request merged**